### ISSUE TITLE
App v2.3.0+ fails to start with an AssertionError

### ISSUE THREAD

**Issue by mss** [Nitrokey/nitrokey-app2#273]
I just installed the Flatpak (version 2.3.0) and tried to start it. It dies with an `AssertionError` right away:

```
# flatpak run com.nitrokey.nitrokey-app2 
Traceback (most recent call last):
  File "/app/bin/nitrokeyapp", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/app/lib/python3.11/site-packages/nitrokeyapp/__main__.py", line 34, in main
    window = GUI(app, log_file)
             ^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/nitrokeyapp/gui.py", line 141, in __init__
    self.init_gui()
  File "/app/lib/python3.11/site-packages/nitrokeyapp/gui.py", line 235, in init_gui
    self.show_device(data)
  File "/app/lib/python3.11/site-packages/nitrokeyapp/gui.py", line 281, in show_device
    view.refresh(data, force=True)
  File "/app/lib/python3.11/site-packages/nitrokeyapp/overview_tab/__init__.py", line 92, in refresh
    assert data.status.variant
           ^^^^^^^^^^^^^^^^^^^
AssertionError
QThread: Destroyed while thread is still running
```

I did not yet install the udev rules as mentioned as step 3 on https://docs.nitrokey.com/software/nk-app2/installation_linux so that might be the issue. If it is, the app should display a proper error message and don't die right away.

I am trying to run this on
```
Operating System: KDE neon 6.0
KDE Plasma Version: 6.1.5
KDE Frameworks Version: 6.5.0
Qt Version: 6.7.2
Kernel Version: 6.8.0-40-generic (64-bit)
Graphics Platform: Wayland
```

**Comment by mss** [comment]
After installing the udev rules the app still crashes with that `AssertionError´. My device is a
```
Bus 001 Device 002: ID 20a0:42b2 Clay Logic Nitrokey 3
```

**Comment by mss** [comment]
I checked v2.2.2 and it doesn't crash but also doesn't like my Nitrokey very much (even though the device itself works):

![image](https://github.com/user-attachments/assets/cb0b63c2-37ca-4382-9aa8-7c0c5b2ddd08)

**Comment by daringer** [comment]
hey, hey - thx for reporting .... This looks like you have an older firmware on the device, therefore some (especially `admin-app`) features are missing - could you run `nitropy nk3 status` and check/paste the firmware version you see there?

**Comment by mss** [comment]
```
Firmware version:   v1.1.0
```

**Comment by daringer** [comment]
totally - let me try to fix that

**Comment by daringer** [comment]
could you test this (https://github.com/Nitrokey/nitrokey-app2/commit/9f6e558fc376e863bcaa21c652a7af07e022e629) with your old firmware ?

**Comment by mss** [comment]
With #282 applied it works as expected:

![image](https://github.com/user-attachments/assets/0b3d0246-423a-4b70-b6c8-c1a57875a037)

**Comment by mss** [comment]
One minor issue: The app should do a rescan after the firmware was updated to make that message go away :-)

**Comment by daringer** [comment]
ahhh, good find, thanks - added an issues here for the next release: https://github.com/Nitrokey/nitrokey-app2/issues/283

**Comment by robin-nitrokey** [comment]
A user reported on Matrix that nitrokey-app2 v2.3.2 still crashed with an old device (shipped in September 2022, so probably v1.2.0).  The exact firmware version and the crash logs are unfortunately not available.

**Comment by daringer** [comment]
yep, I was too optimistic and didn't assert on the fields, just checked their existence - #282 fixes that and will be part of the next release
